First, thanks for the PR!

> and the project compiles 100% cleanly on a Linux environment (Ubuntu 24.04 / GCC 13.3) after this drop-in replacement.

You should really compile on the system that is affected by the changes. In this case, it's macOS, not Linux.

This is the error:
```
/Users/runner/work/supercollider/supercollider/server/scsynth/SC_CoreAudio.cpp:1163:46: error: no member named 'get' in 'std::optional<unsigned int>'
 1163 |         *outSampleRate = mExplicitSampleRate.get();
```
`boost::optional` uses `get()` whereas `std::optional` uses `value()`. Therefore it's not really a drop-in replacement. Just change the code to `*outSampleRate = mExplicitSampleRate.value();`

--- a/server/scsynth/SC_CoreAudio.cpp
+++ b/server/scsynth/SC_CoreAudio.cpp
@@ -1160,7 +1160,7 @@ bool SC_CoreAudioDriver::DriverSetup(int* outNumSamplesPerCallback, double* outS
 
     *outNumSamplesPerCallback = mHardwareBufferSize / outputStreamDesc.mBytesPerFrame;
     if (mExplicitSampleRate) {
-        *outSampleRate = mExplicitSampleRate.get();
+        *outSampleRate = mExplicitSampleRate.value();
     } else {
         *outSampleRate = outputStreamDesc.mSampleRate;
     }

